### PR TITLE
Refactor DataLoader to make typing easier

### PR DESCRIPTION
**Pull request by nikklassen**
I plan to start adding types to components (replacing `any`). This requires establishing a schema for the list items and data queries. Step 1 is here, to break out the data queries to that different queries can accept different types, otherwise they would all be smooshed together and not be very helpful.

You'll also see that I removed some queries from `customQueries.ts`. These appear to be the same as the ones in `queries.ts`, if a little outdated. Currently we don't include `customQueries.ts` in `API.ts`, and we can't because of the conflicts, but this prevents us from using types for those queries.

This exercise found several inconsistencies between the actual data and DataLoader, so I fixed some bugs. For example, staff was using `lastName` instead of `LastName`, and comparing strings using `-`, which won't result in proper sorting, since a string minus a string is `NaN`.

### CODE REVIEW

**Comment by georgevicbell** [comment]
Customqueries was designed to reduce costs and improve perf. It has shallower level of return, which reduces the number of dynamoDB queries if I remember correctly.

Sent from my iPhone

On Jun 17, 2020, at 8:46 AM, Nik Klassen <notifications@github.com> wrote:

﻿

I plan to start adding types to components (replacing any). This requires establishing a schema for the list items and data queries. Step 1 is here, to break out the data queries to that different queries can accept different types, otherwise they would all be smooshed together and not be very helpful.

You'll also see that I removed some queries from customQueries.ts. These appear to be the same as the ones in queries.ts, if a little outdated. Currently we don't include customQueries.ts in API.ts, and we can't because of the conflicts, but this prevents us from using types for those queries.

This exercise found several inconsistencies between the actual data and DataLoader, so I fixed some bugs. For example, staff was using lastName instead of LastName, and comparing strings using -, which won't result in proper sorting, since a string minus a string is NaN.

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/themeetinghouse/web/pull/290

Commit Summary

  *   Refactor DataLoader to make typing easier

File Changes

  *   M .vscode/settings.json<https://github.com/themeetinghouse/web/pull/290/files#diff-5b46cfda0ddaed371f52aa24ea2b0b2a> (10)
  *   A jest.config.js<https://github.com/themeetinghouse/web/pull/290/files#diff-2d0cd5d10b9604941c38c6aac608178a> (8)
  *   M package-lock.json<https://github.com/themeetinghouse/web/pull/290/files#diff-32607347f8126e6534ebc7ebaec4853d> (4835)
  *   M package.json<https://github.com/themeetinghouse/web/pull/290/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2> (3)
  *   A src/__mocks__/aws-amplify.ts<https://github.com/themeetinghouse/web/pull/290/files#diff-49230fd9a2e412ff9a7be36887bbc3aa> (20)
  *   A src/components/RenderRouter/DataLoader.spec.ts<https://github.com/themeetinghouse/web/pull/290/files#diff-3fa85b02d243116eb7a38ec8718cda8a> (661)
  *   M src/components/RenderRouter/DataLoader.ts<https://github.com/themeetinghouse/web/pull/290/files#diff-3b4580d5a843fa711c9d9e7c9ac35410> (651)
  *   M src/components/RenderRouter/HeroItem.tsx<https://github.com/themeetinghouse/web/pull/290/files#diff-7d69913897fc68a00caabfce2c5db348> (91)
  *   M src/components/RenderRouter/HomeChurchItem.tsx<https://github.com/themeetinghouse/web/pull/290/files#diff-4a599c648dd4b21727cd764043f72385> (14)
  *   M src/components/RenderRouter/ListItem.tsx<https://github.com/themeetinghouse/web/pull/290/files#diff-4e914fa03d20987c572793981ebcc8ac> (1245)
  *   M src/components/RenderRouter/WeatherItem.tsx<https://github.com/themeetinghouse/web/pull/290/files#diff-280f37e8badb5e5343902101c4977cd5> (32)
  *   M src/graphql-custom/customQueries.ts<https://github.com/themeetinghouse/web/pull/290/files#diff-b60f6d549eac713965504b7de65898f5> (352)
  *   M src/pages/admin/import-video.tsx<https://github.com/themeetinghouse/web/pull/290/files#diff-c5b4bd5dc9995d421dbdb68fc2d6b81d> (959)
  *   M tsconfig.json<https://github.com/themeetinghouse/web/pull/290/files#diff-e5e546dd2eb0351f813d63d1b39dbc48> (2)

Patch Links:

  *   https://github.com/themeetinghouse/web/pull/290.patch
  *   https://github.com/themeetinghouse/web/pull/290.diff

**Comment by nikklassen** [comment]
They looked the same at a quick inspection, but if that's the case then I recommend renaming the custom one so that we can get the type generation.

**Comment by nikklassen** [comment]
Ok, yup the custom queries look different. I'll fix that. I also created a new commit to just have the formatting changes so this is hopefully easier to review.

File: src/components/RenderRouter/HeroItem.tsx [hunk 1]
```diff
@@ -1,359 +1,557 @@
...
-            title: 'Church at The Meeting House',
-            description: 'Join us at The Meeting House on Sunday!',
-            location: locationItem.location.address,
-            startTime: nextSunday.format(),
-            endTime: moment(nextSunday).add(90, "minutes").format()
-        }
-        return event;
-    }
+  async componentDidMount() {
+    this.getHeight(this.imgUrl(1920) + this.state.content.image1[0].src);
```
Comment (JonEsparaz) [inline src/components/RenderRouter/HeroItem.tsx:38]: This function no longer exists.

Comment (nikklassen) [reply src/components/RenderRouter/HeroItem.tsx]: Good catch, done.

**Comment by JonEsparaz** [comment]
@GeorgeBellTMH have you gotten a chance to look at this? Are we good to merge after the conflicts are resolved?

**Pull request merged**